apps/api/src/app/layouts/usecases/create-layout-change/create-layout-change.use-case.ts

```diff
@@ -13,10 +13,17 @@ export class CreateLayoutChangeUseCase {
   constructor(
     private createChange: CreateChange,
     private findDeletedLayout: FindDeletedLayoutUseCase,
-    private layoutRepository: LayoutRepository
+    private layoutRepository: LayoutRepository,
+    private changeRepository: ChangeRepository
   ) {}
 
   async execute(command: CreateLayoutChangeCommand, isDeleteChange = false): Promise<void> {
+    const parentChangeId: string = await this.changeRepository.getChangeId(
+      command.environmentId,
+      ChangeEntityTypeEnum.LAYOUT,
+      command.layoutId
+    );
```

Comment:
We should move this to inside the condition, to only execute it in case there is a layout found for the different cases.

Reply:
What do you mean by condition?

Reply:
The condition that checks if there is item.

Reply:
Good point